common/interpreter/gipsy_scheme/packages/catch-package.scm

```diff
@@ -27,22 +27,37 @@
      (for-each (lambda (a) (write a)) args)
      (newline))
 
-   (define (throw . x)
+   (define (throw-with-no-continuations . x)
+     (if (_more-handlers?)
+         (begin (apply (_pop-handler) x) (quit -1))
```

Comment:
the (quit ..) seems the crucial "safety-valve" which makes sure that catch-blocks which try to clean-up-and-continue (a use-case not currently supported contrary to catch-and-rethrow) does not continue with non-expected/unpredictable behavior.  It is though easy to overlook and not so trivial to understand: maybe a comment on this and somewhere also some comment on what is (not) supported in the no-continuation-case might help in readability and maintainability?

Reply:
quit is necessary for non-local exit. changed it to be more symmetrical with the continuation form.